core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -313,8 +291,8 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         builder.array(GeoBoundingBoxQueryParser.TOP_LEFT, topLeft.getLon(), topLeft.getLat());
         builder.array(GeoBoundingBoxQueryParser.BOTTOM_RIGHT, bottomRight.getLon(), bottomRight.getLat());
         builder.endObject();
-        builder.field("coerce", coerce);
-        builder.field("ignore_malformed", ignoreMalformed);
+        builder.field("coerce", validationMethod == GeoValidationMethod.COERCE);
+        builder.field("ignore_malformed", validationMethod == GeoValidationMethod.IGNORE_MALFORMED);
```

Comment:
`ignoreMalformed()` would also be true for `GeoValidationMethod.COERCE`, I understand this doesn't change much in terms of the parser logic but maybe it would be more consistent to just use the getters here and rely on the logic they hide.